geps/gep-2648/index.md

```diff
@@ -132,7 +129,35 @@ when a Policy is a Direct Attached Policy:
   of some sort, where the things outside the namespace can opt–out of the behavior.
   (Notably, this is the design that we used for ReferenceGrant).
 
-## Apply Policies to Sections of a Resource
+## Target References
+### Cross Namespace
+
+In all cases, Gateway API policies should only have an effect on the namespace
+they exist within. In the case of policies that could apply to mesh
+implementations, it may be desirable to have a policy that affects traffic
+originating from the local namespace but going to a separate namespace. Unless
+that specific case is desired, all policy target refs should be local and
+exclude the "namespace" field.
+
+### Multiple
+
+In some cases, it may be desirable for a policy to target more than one resource
+at a time. For example, a policy may apply to different variations of what is
+effectively the same Service (store, store-blue, and store-green). If this is
+desired, a policy can choose to support a `targetRefs` list instead of a
+singular `targetRef` field. This list can have a maximum of 16 entries, though
+it may be desirable to start with a lower limit depending on the policy.
+
+#### Migration from Single to Multiple Targets
+
+Existing policies with a single `targetRef` may want to transition to supporting
```

Comment:
happy to see the focus on trying to solve this issue of how a user can author intent once (one policy), and apply it to multiple targets. Pluralizing `targetRef` seems to be the simplest way to support this, but imo this style only works well when the count of `targetRefs` is 2-10. 
When a user wants to define an implementation specific SecurityPolicy resource and apply it to 50 or  even 1000 xRoutes, this becomes tedious. This is where label selectors can help reduce the number of lines to author to get the job done. 
This style is used in other field within Gateway API like [AllowedRoutes](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io%2fv1.RouteNamespaces).

We may want to take smaller steps when supporting multiple targets, but wanted to highlight that there is a real need to support other ways to target resources, and hoping we leaving space for that in the API

Reply:
thinking out loud, here's one way to achieve it
```
targets:
  refs:
  - .....
```

Reply:
+1 for this.

Although matching the policy by labels may be complex and less efficient than the current KV matching used by targetRefs , providing this solution as an optional implementation would be more user-friendly.

Reply:
This has been mentioned _somewhere_ (I believe by @youngnick, but I can't find the reference currently) as a possible future addition if needed, but intentionally omitted to start to keep complexity constrained while we figure out how to handle status and discoverability in a reasonably performant way.

Agreed that `targetRefs` is not ideal for scaling beyond a handful of referenced objects (I think that's intentional and not necessarily a bad thing).

Before considering this addition, I'd first want to dig in to whether using an inherited policy targeted at a namespace or gateway may be sufficient for these use cases rather than targeting dozens or hundreds or routes individually. The best argument I've seen for supporting a selector so far is for https://github.com/kubernetes-sigs/gateway-api/issues/1812, and Istio does already implement a split `targetRef` or `selector` API model on https://istio.io/latest/docs/reference/config/security/authorization-policy/#AuthorizationPolicy and other resources so there are other implementations with prior art, but I'd consider this beyond the scope of this immediate change for now.

Reply:
Agree with @mikemorris that label selection here is a bigger problem than I'm trying to solve in this PR. Likely worth adding this idea to the broader policy discussion in #2927 though to see what kind of traction it gets.

Reply:
@arkodg was part of your original comment here also an argument for starting with a smaller max length than the "16" that I'd originally proposed?

Reply:
@robscott my orig comment was to make sure we can extend this API easily in the future in case we want to add selector support